### PR TITLE
fix: handle correctly multipart stream if the original stream is empty

### LINKED ISSUE #3293: Test: add test with empty stream for multipart protocol

### PR DESCRIPTION
**Pull request by bnjjj**
Fixes #3293

### CODE REVIEW

**Review by Geal — APPROVED** [review]

File: apollo-router/src/protocols/multipart.rs [hunk 1]
```diff
@@ -205,4 +234,35 @@ mod tests {
...
+            } else {
+                match curr_index {
+                    0 => {
+                        assert_eq!(
+                            res,
+                            "\r\n--graphql\r\ncontent-type: application/json\r\n\r\n{}\r\n--graphql--\r\n"
+                        );
+                    }
+                    _ => {
+                        panic!("should not happened, test failed");
```
Comment (o0Ignition0o) [inline apollo-router/src/protocols/multipart.rs:261]:
```suggestion
                        panic!("should not happen, test failed");
```

**Review by o0Ignition0o — APPROVED** [review]
Let's make sure it works end2end with a client, otherwise lgtm!
